webflow-hacks/cms-hacks/58/58.ts

```diff
@@ -0,0 +1,82 @@
+// inject the youtube api script
+const tag = document.createElement('script');
+tag.src = 'https://www.youtube.com/iframe_api';
+const firstScriptTag = document.getElementsByTagName('script')[0];
+if (firstScriptTag.parentNode) {
+  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+}
+
+/**
+ * default function called by the youtube api
+ */
+function onYouTubeIframeAPIReady() {
+  const IFRAME_SELECTOR = '[fs-hacks-element="iframe"]';
+  const iframeContainer = document.querySelector(IFRAME_SELECTOR);
+  if (!iframeContainer) return;
+  const iframes = iframeContainer.querySelectorAll<HTMLIFrameElement>('iframe');
+  iframes.forEach((iframe, index) => {
+    // set iframe properties
+    let src = iframe.getAttribute('src');
+    if (!src) return;
+    if (!src.includes('youtube')) return;
+    const enablejsapiString = src.includes('?') ? '&enablejsapi=1' : '?enablejsapi=1';
+    src = `${src + enablejsapiString}`;
+    iframe.setAttribute('src', src);
+    iframe.setAttribute('id', 'dynamic' + index);
+
+    createPlayer(iframe.id);
+  });
+}
+
+/**
+ * Set up the player, add listeners, and play the video.
+ * @param id: id of the iframe
+ */
+function createPlayer(id: string) {
+  // initialize YT.player with the specified iframe's id
+  const player = new YT.Player(id, {
+    events: {
+      onReady: onPlayerReady,
+    },
+  });
+
+  function onPlayerReady() {
+    const timestampLinks = document.querySelectorAll<HTMLAnchorElement>('.hacks-rich-text a');
+    timestampLinks.forEach((timestampLink) => {
+      timestampLink.addEventListener('click', function (e) {
+        e.preventDefault();
+        const timestamp = timestampLink.innerText.trim();
+        const seconds = convertTimestampToSeconds(timestamp);
+        player.seekTo(seconds, true);
+      });
+    });
+  }
+}
+
+/**
+ * Convert timestamp to seconds
+ * @param timestamp
+ * @returns {number}
+ */
+const convertTimestampToSeconds = (timestamp: string): number => {
+  const timeStampArray = timestamp.split(':');
+  let seconds = 0;
+  timeStampArray.reverse();
+  timeStampArray.forEach((timeStamp, index) => {
+    const timeStampNumber = Number(timeStamp);
+    if (isNaN(timeStampNumber)) return 0;
+    // add seconds
+    if (index === 0) {
+      seconds += timeStampNumber;
+    }
+    // add minutes
+    if (index === 1) {
+      seconds += timeStampNumber * 60;
+    }
+    // add hours
+    if (index === 2) {
+      seconds += timeStampNumber * 60 * 60;
+    }
+  });
+  return seconds;
+};
```

Comment:
Question: have you checked if there's an easier way to do this by using a native `Date` method?

Reply:
did not do research on this but after doing it, I found a more elegant way:
`timestamp.split(':').reduce((acc,time) => (60 * acc) + +time);`
This is even better than date because I don't have two separate statements to parse timestamps that contain hours and those that don't

Reply:
Nice!